### PR TITLE
Inherit

### PR DESCRIPTION
**Pull request by mihaipotoroaca**
tema inherit

### CODE REVIEW

File: new_coffee/coffee.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class Coffee:
+    """
+    Coffee class with prepared coffee quantity and time
+    """
+    def __init__(self, coffee_spoons, sugar, water):
+        self.coffee_spoons = coffee_spoons
+        self. sugar = sugar
```
Comment (flavia-c) [inline new_coffee/coffee.py:7]: space should not exist between self and sugar

File: new_latte/coffee_latte.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+class CoffeeLatte(Coffee):
+    """
+    Coffee class with a bit of milk
+    """
+    def __init__ (self, latte_spoons , latte_sugar, latte_water, milk):
+       super().__init__(coffee_spoons=latte_spoons, sugar=latte_sugar, water=latte_water)
+       self.milk = milk
+
+    def prepare_coffee(self):
+        return Coffee.quantity + self.milk
```
Comment (flavia-c) [inline new_latte/coffee_latte.py:12]: this may overwrite the inherited method if not using super()

File: new_latte/coffee_latte.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    def prepare_time_coffee(self):
+        time_to_prepare = self.water/10 + (self.coffee_spoons*10 + self.sugar*5)/5 + self.milk/10
+        print(f'This coffe latte takes about {time_to_prepare} seconds to prepare')
```
Comment (flavia-c) [inline new_latte/coffee_latte.py:16]: this may overwrite the inherited method if not using super()

File: new_coffee/coffee.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        self.water = water
+    quantity = 0
```
Comment (CosminTraistaru) [inline new_coffee/coffee.py:9]: variables declaration should be done before methods declarations,  right after the class's doc string.

File: new_coffee/coffee.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    first_coffee = Coffee(2,2,40)
+    second_coffee = Coffee(3,1,20)
+
+    first_coffee.prepare_coffee()
+    first_coffee.show_coffee_preparation()
+    first_coffee.prepare_time_coffee()
+
+    second_coffee.prepare_coffee()
+    second_coffee.show_coffee_preparation()
+    second_coffee.prepare_time_coffee()
```
Comment (CosminTraistaru) [inline new_coffee/coffee.py:38]: what will happen if you call `first_coffee.show_coffee_preparation()` at the end?

File: class_incremets.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+class Car:
+    """
+    This is a car class
+    """
+    nrOfInst = 0
```
Comment (CosminTraistaru) [inline class_incremets.py:5]: Use the function naming rules: lowercase with words separated by underscores as necessary to improve readability.
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables

File: class_incremets.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    def __init__(self, name, color):
+        self.name = name
+        self.color = color
+        Car.nrOfInst += 1
+
+    def show(self):
+        print(f'This is a {self.color} car and  is named {self.name}')
+    @staticmethod
+    def ShowNrOfInst():
```
Comment (CosminTraistaru) [inline class_incremets.py:14]: Use the function naming rules: lowercase with words separated by underscores as necessary to improve readability.
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables

File: new_latte/coffee_latte.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+from new_coffee.coffee import *
```
Comment (CosminTraistaru) [inline new_latte/coffee_latte.py:1]: Wildcard imports (from <module> import *) should be avoided
https://www.python.org/dev/peps/pep-0008/#imports

File: new_latte/coffee_latte.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+    def prepare_time_coffee(self):
+        time_to_prepare = self.water/10 + (self.coffee_spoons*10 + self.sugar*5)/5 + self.milk/10
+        print(f'This coffe latte takes about {time_to_prepare} seconds to prepare')
+    def energy(self):
+        energy_level = self.milk * Coffee.quantity**2
+        print(f'This Coffe Latte just raised your energy level with {energy_level}%')
+
+if __name__ == '__main__':
+    first_latte = CoffeeLatte(2,1,30,20)
```
Comment (CosminTraistaru) [inline new_latte/coffee_latte.py:22]: https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements

**Pull request merged**